--- libstdc++-v3/src/c++17/Makefile.in.orig	2022-01-17 17:37:16.470000000 -0600
+++ libstdc++-v3/src/c++17/Makefile.in	2022-01-17 17:37:11.663000000 -0600
@@ -455,7 +455,7 @@
 # OPTIMIZE_CXXFLAGS on the compile line so that -O2 can be overridden
 # as the occasion calls for it.
 AM_CXXFLAGS = \
-	-std=gnu++17 \
+	-std=gnu++17 -nostdinc++ \
 	$(glibcxx_lt_pic_flag) $(glibcxx_compiler_shared_flag) \
 	$(XTEMPLATE_FLAGS) $(VTV_CXXFLAGS) \
 	$(WARN_CXXFLAGS) $(OPTIMIZE_CXXFLAGS) $(CONFIG_CXXFLAGS) \
