From 6fc2b4419ee173e44c1c5b4529772bc8844609dd Mon Sep 17 00:00:00 2001
From: Diego Magdaleno <diegomagdaleno@protonmail.com>
Date: Sat, 8 Jan 2022 14:01:25 -0600
Subject: [PATCH] Experimental Utopia in linux-common

---
 gcc/config/i386/linux-common.h | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/gcc/config/i386/linux-common.h b/gcc/config/i386/linux-common.h
index efa7fb2072c2e..27972a81b9e7a 100644
--- a/gcc/config/i386/linux-common.h
+++ b/gcc/config/i386/linux-common.h
@@ -24,16 +24,20 @@ along with GCC; see the file COPYING3.  If not see
     {                                          \
       GNU_USER_TARGET_OS_CPP_BUILTINS();       \
       ANDROID_TARGET_OS_CPP_BUILTINS();	       \
+      UTOPIA_TARGET_OS_CPP_BUILTINS();         \
     }                                          \
   while (0)
 
 #define EXTRA_TARGET_D_OS_VERSIONS()		\
-  ANDROID_TARGET_D_OS_VERSIONS();
+  ANDROID_TARGET_D_OS_VERSIONS();  \
+  UTOPIA_TARGET_D_OS_VERSIONS();
 
 #undef CC1_SPEC
 #define CC1_SPEC \
   LINUX_OR_ANDROID_CC (GNU_USER_TARGET_CC1_SPEC, \
-		       GNU_USER_TARGET_CC1_SPEC " " ANDROID_CC1_SPEC)
+		       GNU_USER_TARGET_CC1_SPEC " " ANDROID_CC1_SPEC) \
+  LINUX_OR_UTOPIA_CC  (GNU_USER_TARGET_CC1_SPEC, \
+           GNU_USER_TARGET_CC1_SPEC " " UTOPIA_CC1_SPEC)
 
 #undef	LINK_SPEC
 #define LINK_SPEC \
@@ -43,7 +47,9 @@ along with GCC; see the file COPYING3.  If not see
 #undef  LIB_SPEC
 #define LIB_SPEC \
   LINUX_OR_ANDROID_LD (GNU_USER_TARGET_LIB_SPEC, \
-		    GNU_USER_TARGET_NO_PTHREADS_LIB_SPEC " " ANDROID_LIB_SPEC)
+		    GNU_USER_TARGET_NO_PTHREADS_LIB_SPEC " " ANDROID_LIB_SPEC) \
+  LINUX_OR_UTOPIA_LD (GNU_USER_TARGET_LIB_SPEC,  \
+        GNU_USER_TARGET_NO_PTHREADS_LIB_SPEC " " UTOPIA_LIB_SPEC)
 
 #undef  STARTFILE_SPEC
 #define STARTFILE_SPEC \
